http-netty/src/main/java/io/micronaut/http/netty/websocket/AbstractNettyWebSocketHandler.java

```diff
@@ -315,9 +321,31 @@ protected void handleWebSocketFrame(ChannelHandlerContext ctx, WebSocketFrame ms
                         CloseReason.UNSUPPORTED_DATA
                 );
             } else {
+                ByteBuf msgContent = msg.content().retain();
```

Comment:
In general to avoid memory leaks this PR is going to also have to cleanup the `frameBuffer` reference inside `handlerRemoved` and possibly `exceptionCaught` otherwise this could trigger a memory leak I fear